status/sequencing.py

```diff
@@ -0,0 +1,266 @@
+""" Handlers related to data sequencing statistics.
+"""
+from collections import defaultdict
+import cStringIO
+from datetime import datetime
+import json
+
+from dateutil import parser
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_agg import FigureCanvasAgg
+import numpy as np
+import tornado.web
+
+from status.util import dthandler, SafeHandler
+
+
+def make_instrument_series_handler(couchdb_view_name):
+    """ Create a handler for a flowcell-instrument series of data for the
+    given couchdb view.
+
+    Loaded through:
+    /api/v1/instrument_* urls
+    """
+    class InstrumentSeriesDataHandler(SafeHandler):
+        def get(self):
+            self.set_header("Content-type", "application/json")
+            self.write(json.dumps(self.data(), default=dthandler))
+
+        def data(self):
+            view = self.application.flowcells_db.view(couchdb_view_name)
+
+            data = defaultdict(lambda: defaultdict(list))
+            for row in view:
+                date = row.key[0]
+                if not date: # Handles null values
+                    continue
+                instrument = row.key[1].upper()
+                data[instrument][date] += row.value
+
+            return data
+
+    return InstrumentSeriesDataHandler
+
+
+InstrumentClusterDensityDataHandler = make_instrument_series_handler("instrument/clusters_raw")
+
+
+class InstrumentClusterDensityPlotHandler(InstrumentClusterDensityDataHandler):
+    """ Gives a plot for series of cluster densities for instruments, over time.
+    Loaded through /api/v1/instrument_cluster_density.png url
```

Comment:
This one also looks like an old script that has been previously deleted I think..?